### PR TITLE
Jtcarrasco

### PR DESCRIPTION
**Pull request by jtcarrasco**

### CODE REVIEW

File: keyboards/idobo/keymaps/Revok75/keymap.c [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+ * |--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+-----------------+--------|
+ * | CAP LK | A      | S      | D      | F      | G      | H     | J      | K      | L      | ;      | '      | ENTER  | ENTER | REF      |
+ * |--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------------------------+--------|
+ * | LSHIFT | Z      | X      | C      | V      | B      | N     | M      | ,      | .      | /      | \      | RSHIFT| UP     | SAVE     |
+ * |--------+--------+--------+--------+--------+-----------------+--------+--------+--------+--------+-----------------+--------+--------|
+ * |  LALT   | LGUI   | LALT	|	FIND	| CMD T   | SPACE  | SPACE | ALFRED | FN-MO  | RALT   | RGUI   | RCTRL  | LEFT  | DOWN   | RIGHT    |
+ * '--------------------------------------------------------------------------------------------------------------------------------------'
+ */
+
+ [_QW_M] = { /* QWERTY MAC*/
```
Comment (noroadsleft) [inline keyboards/idobo/keymaps/Revok75/keymap.c]: All your layers should be formatted to use a layout macro, like this:

https://github.com/qmk/qmk_firmware/blob/87e6d0137484fdc8e076eb79dbbed47836eba2c5/keyboards/idobo/keymaps/default/keymap.c#L25-L31

File: keyboards/idobo/keymaps/Revok75/keymap.c [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+ [_FN] = { /* OSLAYOUT + NUMPAD + MEDIA + LIGHTING */
+   { KC_TRNS, KC_F1,     KC_F2,						 KC_F3,    		   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_TRNS, KC_MUTE, KC_VOLD, KC_VOLU},
+   { KC_TRNS, KC_NO,     DF(_QW_W),    		 KC_NO,    		   KC_NO,   RGB_TOG, KC_NO,   KC_NO,   KC_RALT, RESET,   KC_NO,   KC_LBRC, KC_RBRC, KC_TRNS, KC_TRNS},
+   { KC_TRNS, DF(_QW_M), RGB_MODE_RAINBOW, RGB_MODE_PLAIN, KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS},
+   { KC_TRNS, RGB_VAI,   RGB_VAD,  				 RGB_HUI,  		   RGB_HUD, RGB_SAI, RGB_SAD, KC_NO,   KC_NO,   KC_DOT,  KC_TRNS, KC_TRNS, KC_TRNS, KC_HOME, KC_TRNS},
+   { KC_TRNS, KC_TRNS,   KC_TRNS,  				 KC_TRNS,  		   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_PGUP, KC_END,	 KC_PGDN},
+ }
+};
+
+const uint16_t PROGMEM fn_actions[] = {
```
Comment (noroadsleft) [inline keyboards/idobo/keymaps/Revok75/keymap.c]: Everything from Line 92 to the end of the file can be safely deleted.

`fn_actions` and `action_get_macro` are both deprecated, and your keymap isn't using either function.

File: keyboards/xd75/keymaps/Revok75/config.h [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef CONFIG_USER_H
```
Comment (noroadsleft) [inline keyboards/xd75/keymaps/Revok75/config.h]: If you could replace the include guard (the ifndef, define, and endif at the end) with just `#pragma once`?

It's simpler, and less prone to user error, so if you could make this switch?

```suggestion
#pragma once
```

File: keyboards/xd75/keymaps/Revok75/config.h [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+#define CONFIG_USER_H
```
Comment (noroadsleft) [inline keyboards/xd75/keymaps/Revok75/config.h]:
```suggestion
```

File: keyboards/xd75/keymaps/Revok75/config.h [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+#include "../../config.h"
+// place overrides here
+#undef BACKLIGHT_BREATHING
+
+#endif
```
Comment (noroadsleft) [inline keyboards/xd75/keymaps/Revok75/config.h]:
```suggestion
```

File: keyboards/xd75/keymaps/Revok75/config.h [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef CONFIG_USER_H
+#define CONFIG_USER_H
+
+#include "../../config.h"
```
Comment (noroadsleft) [inline keyboards/xd75/keymaps/Revok75/config.h]: Line 20 can be safely deleted. QMK's build script automatically includes all the `config.h` files in the build tree, so this line is redundant.

```suggestion
```

File: keyboards/xd75/keymaps/Revok75/keymap.c [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "xd75.h"
```
Comment (noroadsleft) [inline keyboards/xd75/keymaps/Revok75/keymap.c]: This line should be:

```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/xd75/keymaps/Revok75/keymap.c [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+ * |--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+-----------------+--------|
+ * | CAP LK | A      | S      | D      | F      | G      | H     | J      | K      | L      | ;      | '      | ENTER  | ENTER | REF      |
+ * |--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------------------------+--------|
+ * | LSHIFT | Z      | X      | C      | V      | B      | N     | M      | ,      | .      | /      | \      | RSHIFT| UP     | SAVE     |
+ * |--------+--------+--------+--------+--------+-----------------+--------+--------+--------+--------+-----------------+--------+--------|
+ * |  LALT   | LGUI   | LALT	|	FIND	| CMD T   | SPACE  | SPACE | ALFRED | FN-MO  | RALT   | RGUI   | RCTRL  | LEFT  | DOWN   | RIGHT    |
+ * '--------------------------------------------------------------------------------------------------------------------------------------'
+ */
+
+ [_QW_M] = { /* QWERTY MAC*/
```
Comment (noroadsleft) [inline keyboards/xd75/keymaps/Revok75/keymap.c]: Again, all your layers should be formatted to use the `LAYOUT_ortho_5x15` layout macro, as described above.

File: keyboards/xd75/keymaps/Revok75/keymap.c [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+ [_FN] = { /* OSLAYOUT + NUMPAD + MEDIA + LIGHTING */
+   { KC_TRNS, KC_F1,     KC_F2,						 KC_F3,    		   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_TRNS, KC_MUTE, KC_VOLD, KC_VOLU},
+   { KC_TRNS, KC_NO,     DF(_QW_W),    		 KC_NO,    		   KC_NO,   RGB_TOG, KC_NO,   KC_NO,   KC_RALT, RESET,   KC_NO,   KC_LBRC, KC_RBRC, KC_TRNS, KC_TRNS},
+   { KC_TRNS, DF(_QW_M), RGB_MODE_RAINBOW, RGB_MODE_PLAIN, KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS},
+   { KC_TRNS, RGB_VAI,   RGB_VAD,  				 RGB_HUI,  		   RGB_HUD, RGB_SAI, RGB_SAD, KC_NO,   KC_NO,   KC_DOT,  KC_TRNS, KC_TRNS, KC_TRNS, KC_HOME, KC_TRNS},
+   { KC_TRNS, KC_TRNS,   KC_TRNS,  				 KC_TRNS,  		   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_PGUP, KC_END,	 KC_PGDN},
+ }
+};
+
+const uint16_t PROGMEM fn_actions[] = {
```
Comment (noroadsleft) [inline keyboards/xd75/keymaps/Revok75/keymap.c]: Again, everything from Line 92 to the end of the file can be deleted.

File: keyboards/xd75/keymaps/Revok75/rules.mk [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+BOOTMAGIC_ENABLE = yes
+BACKLIGHT_ENABLE = yes
+RGBLIGHT_ENABLE = yes
+
+ifndef QUANTUM_DIR
```
Comment (noroadsleft) [inline keyboards/xd75/keymaps/Revok75/rules.mk]: Lines 20-22 can be deleted. This code block is left over from a previous build script.

File: keyboards/idobo/keymaps/Revok75/keymap.c (hunk 1 above)
Comment (jtcarrasco) [reply keyboards/idobo/keymaps/Revok75/keymap.c]: resolved

File: keyboards/idobo/keymaps/Revok75/keymap.c (hunk 2 above)
Comment (jtcarrasco) [reply keyboards/idobo/keymaps/Revok75/keymap.c]: resolved

File: keyboards/xd75/keymaps/Revok75/config.h (hunk 3 above)
Comment (jtcarrasco) [reply keyboards/xd75/keymaps/Revok75/config.h]: resolved

File: keyboards/xd75/keymaps/Revok75/config.h (hunk 5 above)
Comment (jtcarrasco) [reply keyboards/xd75/keymaps/Revok75/config.h]: resolved

File: keyboards/xd75/keymaps/Revok75/keymap.c (hunk 7 above)
Comment (jtcarrasco) [reply keyboards/xd75/keymaps/Revok75/keymap.c]: resolved

File: keyboards/xd75/keymaps/Revok75/keymap.c (hunk 8 above)
Comment (jtcarrasco) [reply keyboards/xd75/keymaps/Revok75/keymap.c]: resolved

File: keyboards/xd75/keymaps/Revok75/keymap.c (hunk 9 above)
Comment (jtcarrasco) [reply keyboards/xd75/keymaps/Revok75/keymap.c]: resolved

File: keyboards/xd75/keymaps/Revok75/rules.mk (hunk 10 above)
Comment (jtcarrasco) [reply keyboards/xd75/keymaps/Revok75/rules.mk]: resoved

File: keyboards/idobo/keymaps/Revok75/keymap.c [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+	  * | LSHIFT | Z      | X      | C      | V      | B      | N     | M      | ,      | .      | /      | \      | RSHIFT | UP    | SAVE     |
+	  * |--------+--------+--------+--------+--------+-----------------+--------+--------+--------+--------+-----------------+--------+--------|
+	  * | LALT   | LGUI   | LALT   |  FIND  | CMD T	 | SPACE | SPACE  | START  | FN-MO  | RALT   | RGUI   | RCTRL  | LEFT  | DOWN   | RIGHT    |
+	  * '--------------------------------------------------------------------------------------------------------------------------------------'
+	  */
+
+  [_QW_W] = LAYOUT_ortho_5x15( /* QWERTY WIN*/
+  	KC_GESC,  KC_1,     KC_2,     KC_3,     		KC_4,     	KC_5, 	KC_6,   KC_7,   				KC_8,     KC_9,     KC_0,     KC_MINS,     KC_EQL,     LCTL(KC_X),  KC_BSPC, \
+  	KC_TAB,   KC_Q,     KC_W,     KC_E,     		KC_R,     	KC_T, 	KC_Y,   KC_U,   				KC_I,     KC_O,     KC_P,     KC_LCBR,     KC_RCBR,    LCTL(KC_C),  LCTL(KC_V), \
+  	KC_CAPS,  KC_A,     KC_S,     KC_D,     		KC_F,     	KC_G, 	KC_H,   KC_J,   				KC_K,     KC_L,     KC_SCLN,  KC_QUOT,     LCTL(KC_R), KC_NO,       KC_ENT}, \
```
Comment (noroadsleft) [inline keyboards/idobo/keymaps/Revok75/keymap.c]: Here's the error that's causing your Idobo keymap to not compile. The closing curly brace on this line didn't get removed.

```suggestion
  	KC_CAPS,  KC_A,     KC_S,     KC_D,     		KC_F,     	KC_G, 	KC_H,   KC_J,   				KC_K,     KC_L,     KC_SCLN,  KC_QUOT,     LCTL(KC_R), KC_NO,       KC_ENT, \
```

File: keyboards/xd75/keymaps/Revok75/keymap.c (hunk 7 above)
Comment (drashna) [reply keyboards/xd75/keymaps/Revok75/keymap.c]: doesn't appear to be resolved yet. May need to push your changes?

**Comment by drashna** [comment]
Also, could you make sure that the folder and files are ALL lower case. 

Since QMK is cross platform, the casing can cause issues on some systems (read: Windows).  So please rename to use lower case. 

You may need to rename it to something else, commit and rename it back and commit again to accomplish this.

File: keyboards/xd75/keymaps/Revok75/rules.mk (hunk 10 above)
Comment (jtcarrasco) [reply keyboards/xd75/keymaps/Revok75/rules.mk]: resolved

File: keyboards/xd75/keymaps/Revok75/keymap.c (hunk 8 above)
Comment (jtcarrasco) [reply keyboards/xd75/keymaps/Revok75/keymap.c]: resolved

File: keyboards/xd75/keymaps/Revok75/config.h (hunk 4 above)
Comment (jtcarrasco) [reply keyboards/xd75/keymaps/Revok75/config.h]: resolved

File: keyboards/idobo/keymaps/Revok75/keymap.c (hunk 1 above)
Comment (jtcarrasco) [reply keyboards/idobo/keymaps/Revok75/keymap.c]: resolved

**Comment by jtcarrasco** [comment]
file names were changed on my branch, not sure why this isn't showing up.

**Comment by jtcarrasco** [comment]
filename changes should be resolved

**Comment by drashna** [comment]
Awesome, thanks!

**Pull request merged**